Make a yearly cost optimization case study with cooling

danielviladrich opened the pull request:
Most of the time Barcelona can be over the maximum temperature of 24C and would need cooling to stay within the feasible temperature range. Since most buildings have similar cooling power, we can implement a cooling element feature as a negative cooling value and modify the cost function to simulate cost savings also during the summer

danielviladrich committed:
revised utils functions for two component lump model

danielviladrich committed:
refactor, added house class to gather parameters, 3D factor analysis plot

danielviladrich committed:
added comparison script

danielviladrich committed:
reworked combined plot

danielviladrich committed:
add constant toll charge to electricity price

danielviladrich committed:
added temperature jump penalty

danielviladrich committed:
fixed plot legend

danielviladrich committed:
resolved merge conflicts

danielviladrich committed:
added extra required packages

danielviladrich committed:
changed version to 3.11

danielviladrich committed:
remove windows requirement

danielviladrich committed:
changed workflow python version back to original 3.12.8

danielviladrich committed:
made example parameters more explicit

danielviladrich committed:
updated baseline_compare

danielviladrich committed:
added dataframe regularizer, row number number is not constant = 24 anymore

danielviladrich committed:
split example scenarios

danielviladrich committed:
delete

danielviladrich committed:
renamed temperature to outdoor temperature

danielviladrich committed:
added fixed toll and tax price

danielviladrich committed:
moved tolls and taxes to house parameter and later functions

danielviladrich committed:
added wall temperature to state, fixed time price error when in minutes

danielviladrich committed:
reworked plot function

danielviladrich committed:
renamed parameters

danielviladrich committed:
vectorized ODE contraints, renamed heater_state to heater_output

danielviladrich committed:
fix

danielviladrich committed:
renamed heater state and decision to output

danielviladrich committed:
renamed parameter, vectorized ODE constraints

danielviladrich committed:
added time_range option and tests

danielviladrich committed:
slimmed down requirements

danielviladrich committed:
remove base price to avoid failing cost tests

danielviladrich committed:
updated gitignore, requirements

danielviladrich committed:
added pytest install to config

danielviladrich committed:
added extra packages

danielviladrich committed:
revert back to old setup

danielviladrich committed:
added python kasa package

danielviladrich committed:
removed packages

danielviladrich committed:
update gitignore

danielviladrich committed:
Merge branch 'historical-data' of https://github.com/Bloodwing1/Strom into historical-data

danielviladrich committed:
function refactor

danielviladrich committed:
added draft get_historical_weather_data function

danielviladrich committed:
data wrangling refactor

danielviladrich committed:
remove historical weather function

danielviladrich committed:
steamlined historical comparison files

danielviladrich committed:
rename and edit comparison file

danielviladrich committed:
simplify regularization

danielviladrich committed:
fix bug in price units

danielviladrich committed:
better regular interval check

danielviladrich committed:
added variable frequency

danielviladrich committed:
smaller P_base

danielviladrich committed:
separated plot_utils

danielviladrich committed:
more space

danielviladrich committed:
edit

danielviladrich committed:
change save name

danielviladrich committed:
removed obsolete get_state function

danielviladrich committed:
added freq parameter

danielviladrich committed:
cleaner dataframe wrangling

danielviladrich committed:
add missing frequency resampling

danielviladrich committed:
make verbose default

danielviladrich committed:
switch back to original november timeframe to avoid unfeasible high temperatures

danielviladrich committed:
added format handling flags

danielviladrich committed:
changed to linear interpolation to avoid errors when solving

danielviladrich committed:
added prepush hook

danielviladrich committed:
edit

danielviladrich committed:
try again

danielviladrich committed:
edit

danielviladrich committed:
removed test

danielviladrich committed:
added cooling parameter and adapted optimization

danielviladrich committed:
renamed and reworked historical analysis scripts

danielviladrich committed:
remove old cooling unit parameter

danielviladrich committed:
rearrange

danielviladrich committed:
get historical data script

danielviladrich committed:
remove optimization done in Julia

danielviladrich committed:
rework get_temp_price_from_temp function

danielviladrich committed:
Dataframe Column reaname 1

danielviladrich committed:
plot_state removed

danielviladrich committed:
plot_state removed 2

danielviladrich committed:
Dataframe column rename 2 and plot comparison refactor

danielviladrich committed:
savefig rename

danielviladrich committed:
add cooling output

danielviladrich committed:
update compare scritps

danielviladrich committed:
smaller fontsize

danielviladrich committed:
new parameter value

danielviladrich committed:
adjust parameter

danielviladrich committed:
add a small amount of cost sensitivity to avoid unrealistic heater + cooling scenarios

danielviladrich committed:
remove time frame until api calls into the past can be done

danielviladrich committed:
file cleanup

janbsc4 reviewed (changes requested):
Thank you @dviladrich95! The code is currently not passing the automated tests in Github. If you click on the failed checks you can get a log of the unsuccessful run and troubleshoot it!

danielviladrich committed:
removed outdated test

janbsc4 reviewed (changes requested):
Hihi! I tried to run main.py of this branch directly in my IDE and I run into errors. Is it the same for you @dviladrich95? 
Thanks for fixing the tests not passing!

danielviladrich commented:
> Hihi! I tried to run main.py of this branch directly in my IDE and I run into errors. Is it the same for you @dviladrich95? 
> Thanks for fixing the tests not passing!

Good point, since the main.py script is not too heavy I will incorporate it into the testing script as well

danielviladrich committed:
added new function names

danielviladrich committed:
added docs and type hints

danielviladrich committed:
basic type checking corrections

danielviladrich committed:
standard type checking corrections

danielviladrich committed:
updated main function calls, added house parameter config entry

danielviladrich committed:
added copy of main function into strom directory and main test cases with mock tapo device

danielviladrich committed:
added config directory mkdir to workflow

danielviladrich commented:
@janbsc4 I added a test of the main function, now moved inside of strom. Since the main function interacts with the tapo device I set up a mock device, but its methods might not coincide with the ones of the real case, we should double check by running the actual script and connect with the tapo device

janbsc4 commented:
> @janbsc4 I added a test of the main function, now moved inside of strom. Since the main function interacts with the tapo device I set up a mock device, but its methods might not coincide with the ones of the real case, we should double check by running the actual script and connect with the tapo device

@dviladrich95 nice! have you added a new required file in the config folder? we should document that if its the case.

danielviladrich committed:
added documentation

danielviladrich committed:
added empty json file creation fallback

danielviladrich committed:
removed extra empty json file creation from github workflow, now done by main

danielviladrich committed:
add making the config directory to the workflow

danielviladrich committed:
Merge branch 'year-with-cooling' of https://github.com/Bloodwing1/Strom into year-with-cooling

danielviladrich committed:
add setup function to test script

janbsc4 commented:
Reviewed! Thank you @dviladrich95 ! Can be merged!

janbsc4 reviewed (approved):
Ready to be merged!
